### DISCUSSION TITLE
[@types/request] Error: has no exported member named 'CookieJar'

### DISCUSSION THREAD

**Discussion by thurti** [DefinitelyTyped/DefinitelyTyped#71418] [Issues with a @types package]
I get the following typescript error

```bash

node_modules/@types/request/index.d.ts:389:84 - error TS2724: '"[...]/node_modules/tough-cookie/dist/cookie/index"'   
has no exported member named 'CookieJar'. Did you mean 'Cookie'?

389         setCookie(cookieOrStr: Cookie | string, uri: string | Url, options?: tough.CookieJar.SetCookieOptions): void;
                                                                                       ~~~~~~~~~


Found 1 error in node_modules/@types/request/index.d.ts:389
```

My project contains jsdom which has `"tough-cookie": "^5.0.0"` as a dependency. If I understand correctly, the error occurs because the types are imported from this package instead of `@types/tough-cookie`. 

Changing the code to `options?: tough.SetCookieOptions` fixes the error for me. But I think it would probably be better to use the types from `@types/tough-cookie` somehow.?

[a comment by an automated account is not shown]

**Comment by SametSahin10** [comment]
I'm having the same issue.

**Comment by melod1n** [comment]
Same issue

**Comment by mcintoac-aws** [comment]
Also seeing this issue when running tsc using tough-cookie 5.0.0.

Any updates on a fix?

**Comment by tsawyer999** [comment]
Hello everyone, we got a legacy project assigned to our team and facing the same error:

```
node_modules/tough-cookie/dist/cookie/index"' has no exported member named 'CookieJar'. Did you mean 'Cookie'?

setCookie(cookieOrStr: Cookie | string, uri: string | Url, options?: tough.CookieJar.SetCookieOptions): void;
```

Thank you!

**Comment by ZM-J** [comment]
I encountered the same issue when I tried running a legacy project. Any help would be welcomed.

**Comment by S4Sloth** [comment]
I have the same issue.

It prevented my project from building, so I had to modify `build` script with `--skipLibCheck` flag:

```
// package.json
"build": "tsc --skipLibCheck", 
```

but it's just muting warning, so real fix would be appreciated.

**Comment by dylanwooters** [comment]
Also facing this issue when building a legacy app. Any help would be appreciated.

**Comment by chrisvanderkooi** [comment]
Same thing here, been trying everything under the sun to try and fix this to no avail.

**Comment by SrikanthAlva** [comment]
facing the same issue here as well.. unable to get past this issue.

**Comment by mariaeltsova** [comment]
Facing the same issue after upgrading from from jsdom@24 to jsdom@26 which is using tough-cookie@5.1.1

**Comment by MihaelIsaev** [comment]
Facing the same issue with `jsdom ` please help

**Comment by Pastbin** [comment]
"skipLibCheck": true - tsconfig

**Comment by maybeSwarup** [comment]
`tough.CookieJar` is no longer exported from tough-cookie@5.x.x
installing a previous version solved my problem
```
npm i tough-cookie@4.1.4
```

**Comment by hkleungai** [comment]
I have submitted PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/72413 to fix this issue. You may want to have a look.

**Reply by maybeSwarup** [reply]
great, thanks!

**Reply by tsawyer999** [reply]
Thank you @hkleungai!

**Comment by mhuynh5757** [comment]
Hey all, seems like this issue is still persistent. For now, we have an override in our `package.json` to pin `tough-cookie` to `4.1.4`, but this seems like an unsustainable fix for the long term. Can we get some advice on a proper fix, or is there another PR that is working to address this? It seems like from the PR linked above there was an alternate fix that the DefinitelyTyped team recommended, but it's not clear what the status is on that.

**Comment by freben** [comment]
Just one more +1 for this problem. For us, I preferred to pin `"@types/request": "2.0.0"` (many years old) since it didn't have the offending `@types/tough-cookie` dep at all, and yet didn't lead to any other tsc issues. Could perhaps have yarn patched the `package.json` or something instead, but this was at the level of "just make it work", to move the offending thing out of the way.

**Comment by marekdon** [comment]
One more +1. Now I have problem with angular 22, airbrake/browser and @types/request.
